lib/collection/src/collection/mod.rs

```diff
@@ -50,6 +51,7 @@ pub struct Collection {
     this_peer_id: PeerId,
     path: PathBuf,
     snapshots_path: PathBuf,
+    snapshot_storage_manager: Box<dyn SnapshotStorage>,
```

Comment:
This could be static dispatch by making Collection generic over SnapshotStorage. We could switch to it if snapshotting happens often because it is cheaper.